test/feature-flags/index.js

```diff
@@ -0,0 +1,296 @@
+  });
+
+  it('throws if interval is zero', () => {
+    assert.throws(() => initialise({ interval: 0 }, log));
+  });
+
+  it('throws if interval is NaN', () => {
+    assert.throws(() => initialise({ interval: NaN }, log));
+  });
+
+  it('throws if interval is Infinity', () => {
+    assert.throws(() => initialise({ interval: Number.POSITIVE_INFINITY }, log));
+    assert.throws(() => initialise({ interval: Number.NEGATIVE_INFINITY }, log));
+  });
+
+  it('throws if log argument is missing', () => {
+    assert.throws(() => initialise({ interval: 300000 }));
+  });
+
+  describe('initialise, successful get:', () => {
+    let featureFlags;
+
+    beforeEach(done => {
+      featureFlags = initialise({
+        interval: 300000,
+        redis: {
+          enabled: false,
+          host: '127.0.0.1',
+          port: 6379,
+          prefix: 'wibble:',
+        }
+      }, log);
+      resolve(JSON.stringify({ foo: 'bar' }));
+      setImmediate(done);
+    });
+
+    it('returned the expected interface', () => {
+      assert.isObject(featureFlags);
+      assert.isFunction(featureFlags.get);
+      assert.lengthOf(featureFlags.get, 0);
+      assert.isFunction(featureFlags.terminate);
+      assert.lengthOf(featureFlags.terminate, 0);
+    });
+
+    it('initialised redis', () => {
+      assert.equal(redisFactory.callCount, 1);
+      const args = redisFactory.args[0];
+      assert.lengthOf(args, 2);
+      assert.deepEqual(args[0], {
+        enabled: true,
+        host: '127.0.0.1',
+        port: 6379,
+        prefix: 'featureFlags:',
+      });
+      assert.equal(args[1], log);
+    });
+
+    it('called redis.get', () => {
+      assert.equal(redis.get.callCount, 1);
+      const args = redis.get.args[0];
+      assert.lengthOf(args, 1);
+      assert.equal(args[0], 'current');
+    });
+
+    it('called setTimeout', () => {
+      assert.equal(setTimeout.callCount, 1);
+      const args = setTimeout.args[0];
+      assert.lengthOf(args, 2);
+      assert.isFunction(args[0]);
+      assert.equal(args[1], 300000);
+    });
+
+    it('featureFlags.get resolves', async () => {
+      let result;
+      try {
+        result = await featureFlags.get();
+      } catch (err) {
+      }
+      assert.deepEqual(result, { foo: 'bar' });
+    });
+
+    it('did not call clearTimeout', () => {
+      assert.equal(clearTimeout.callCount, 0);
+    });
+
+    describe('terminate:', () => {
+      beforeEach(() => {
+        return featureFlags.terminate();
+      });
+
+      it('called clearTimeout', () => {
+        assert.equal(clearTimeout.callCount, 1);
+        const args = clearTimeout.args[0];
+        assert.lengthOf(args, 1);
+        assert.equal(args[0], 'wibble');
+      });
+
+      it('called redis.close', () => {
+        assert.equal(redis.close.callCount, 1);
+        assert.lengthOf(redis.close.args[0], 0);
+      });
+
+      describe('terminate:', () => {
```

Comment:
What state are you trying to test below? If you called `.terminate()` twice library handles it correctly?

Reply:
Yep. There's no need to call `clearTimeout` twice, so I figured I'd assert that we don't.